DEV: Update linting

cvx opened the pull request.

nbianca commented on app/controllers/multilingual/admin_translations_controller.rb:
```diff
@@ -22,18 +22,19 @@ def upload
 
-        file = Multilingual::TranslationFile.new(opts)
+        opts = process_filename(file)
+          raise opts[:error] if opts[:error]
```
```suggestion
        raise opts[:error] if opts[:error]
```

Drenmi committed:
DEV: Pass respect_plugin_enabled to add_to_serializer as keyword argument (#2)

Passing the respect_plugin_enabled argument to #add_to_serializer as a positional argument is deprecated. It is now expected to be a keyword argument. This PR adds the keyword.

nattsw committed:
FIX: Failing tests due to i18n.default (#3)

nattsw committed:
DEV: Update tests for locale detection and topic creation (#7)

This change was introduced in
- https://github.com/discourse/discourse/pull/24242
- https://github.com/discourse/discourse/pull/24257
and so we have to update the plugin.

davidtaylorhq committed:
DEV: Resolve link-to.positional-arguments deprecation (#8)

* DEV: Resolve link-to.positional-arguments deprecation

https://deprecations.emberjs.com/v3.x#toc_ember-glimmer-link-to-positional-arguments

This deprecation happens at build-time, which means it doesn't get surfaced like other deprecations in Discourse themes/plugins.

* DEV: Fix tests

---------

Co-authored-by: Jarek Radosz <jradosz@gmail.com>

cvx committed:
DEV: Update linting

cvx committed:
Merge branch 'main' into linting

cvx committed:
remove yarn.lock from gitignore

cvx committed:
add yarn.lock

cvx committed:
remove travis config
